### PR TITLE
Retirement roadmap for pytest-openfiles

### PR DESCRIPTION
**Pull request by pllim**
This roadmap goes with

* https://github.com/astropy/astropy/pull/14041

Also see

* https://github.com/astropy/pytest-astropy/pull/52

cc @jdavies-st

### CODE REVIEW

**Comment by pllim** [comment]
I added this to the proposed agenda for infrastructure tag-up.

**Comment by astrofrog** [comment]
Could you add details about what to use instead?

**Comment by pllim** [comment]
I added https://docs.astropy.org/en/latest/development/testguide.html#testing-for-open-files to the roadmap doc here.

File: README.rst
Comment (jdavies-st) [inline README.rst]: I think it would be good to have an example here in this repo that is concrete.  I.e.:


For open file testing, use Python's built in warning-to-error handling switch `-W`.  Pytest invocations support this as well.  To capture all open file ResourceWarnings:

```
pytest -W error::ResourceWarning
```

or set it up in your setup.cfg:

```
[tool:pytest]
filterwarnings =
    error::ResourceWarning
```

For more detail on `-W`, see https://docs.python.org/3/using/cmdline.html#cmdoption-W

Or something similar.

Comment (pllim) [reply README.rst]: OK done. Thanks for the suggestion!

**Comment by bsipocz** [comment]
This needs to be removed from the install list of pytest-astropy, too

**Comment by pllim** [comment]
> This needs to be removed from the install list of pytest-astropy, too

@bsipocz , good point! When would be a good time to do that? I have opened https://github.com/astropy/pytest-astropy/issues/51

**Comment by bsipocz** [comment]
>> This needs to be removed from the install list of pytest-astropy, too

> @bsipocz , good point! When would be a good time to do that? I have opened https://github.com/astropy/pytest-astropy/issues/51

I think the order of things should be:

1a) remove usage. It's already done for core, but maybe do the good old batch-issue for affiliated packages (or at least for the subset of them that use `pytest-astropy`). Maybe even open issues for any packages that do `pytest --open-files`? There are many out there, outside of astronomy who use these plugins (hopefully not open-files).
1b) remove it from `pytest-astropy`. Can be done asap, but need to communicate it wider than the astropy slack and mailing list.
2) wait
3) archive this repo

**Comment by pllim** [comment]
Re: batch issue -- Great idea! Though maybe want to wait until this is merged first.

**Review by astrofrog — APPROVED** [review]
This is good to go!

**Pull request merged**
